libcxx/include/__functional/function.h

```diff
@@ -257,84 +176,38 @@ class __base<_Rp(_ArgTypes...)> {
 
 // __func implements __base for a given functor type.
 
-template <class _FD, class _Alloc, class _FB>
+template <class _FD, class _FB>
 class __func;
 
-template <class _Fp, class _Alloc, class _Rp, class... _ArgTypes>
-class __func<_Fp, _Alloc, _Rp(_ArgTypes...)> : public __base<_Rp(_ArgTypes...)> {
-  __alloc_func<_Fp, _Alloc, _Rp(_ArgTypes...)> __f_;
+template <class _Fp, class _Rp, class... _ArgTypes>
+class __func<_Fp, _Rp(_ArgTypes...)> : public __base<_Rp(_ArgTypes...)> {
+  _Fp __func_;
```

Comment:
I reckon this is not an ABI break because `std::function` is type erased. Hence, anyone with a `std::function<R(Args..)>` on their ABI boundary will access the underlying callable object via its vtable, which does the right thing. So if an existing executable is producing an old-style `__func<_Fp, _Alloc, _Rp(_ArgTypes...)>` object, it'll still work with the current implementation of `std::function`, even though the same source code would now instead lead to a new-style `__func<_Fp, _Rp(_ArgTypes...)>` object being created if recompiled.